reconcile/vpc_peerings_validator.py

```diff
@@ -14,10 +16,136 @@
 )
 from reconcile.status import ExitCodes
 from reconcile.utils import gql
+from reconcile.utils.aws_api import AWSApi
 
 QONTRACT_INTEGRATION = "vpc-peerings-validator"
 
 
+def validate_no_cidr_overlap(
+    query_data: VpcPeeringsValidatorQueryData,
+) -> bool:
+    clusters: list[ClusterV1] = query_data.clusters or []
+
+    for cluster in clusters:
+        peerings_enteries_dict = {}  # type: ignore[var-annotated]
```

Comment:
We would init peerings_enteries with cluster self vpc first.
```suggestion
        peerings_enteries = [
            {
                "provider": "cluster-self-vpc",
                "vpc_name": cluster.name,
                "cidr_block": cluster.network.vpc,
            },
        ] 
```